plaso/parsers/gdrive_synclog.py

```diff
@@ -0,0 +1,197 @@
+    # Initialize it without the TZ, then convert to a TZ-aware value with
+    # a formatting hack and CopyFromDateTimeString.
+    dfdatetime_obj = dfdatetime_time_elements.TimeElementsInMilliseconds(
+        (year, month, day_of_month, hours, minutes, seconds, milliseconds))
+
+    dt_hack = ('{0:d}-{1:02d}-{2:02d} {3:02d}:'
+               '{4:02d}:{5:02d}.{6:03d}{7!s}:{8!s}').format(
+                   year, month, day_of_month, hours, minutes, seconds,
+                   milliseconds, tz[0:3], tz[3:5])
+    try:
+      dfdatetime_obj.CopyFromDateTimeString(dt_hack)
+    except ValueError:
+      parser_mediator.ProduceExtractionError(
+          'invalid date time value: {0!s}'.format(structure.date_time))
+      return
+
+    event_data = GoogleDriveSyncLogEventData()
+    event_data.log_level = structure.log_level
+    event_data.pid = structure.pid
+    event_data.thread = structure.thread
+    event_data.source_code = structure.source_code
+    # Replace newlines with spaces in structure.message to preserve output.
+    event_data.message = structure.message.replace('\n', ' ')
+
+    event = time_events.DateTimeValuesEvent(
+        dfdatetime_obj, definitions.TIME_DESCRIPTION_ADDED)
+
+    parser_mediator.ProduceEventWithEventData(event, event_data)
+
+  def ParseRecord(self, parser_mediator, key, structure):
+    """Parses a log record structure and produces events.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+      key (str): identifier of the structure of tokens.
+      structure (pyparsing.ParseResults): structure of tokens derived from
+          a line of a text file.
+
+    Raises:
+      ParseError: when the structure type is unknown.
+    """
+    # Note: this may be overcomplicated/unnecessary; I'm mostly cloning SkyDrive
+    # parser but Google Drive sync seems to have no header/other key types:
+    # consider refactoring or extending this in the future.
+    if key == 'logline':
+      self._ParseLine(parser_mediator, structure)
+    else:
+      raise errors.ParseError(
+          'Unable to parse record, unknown structure: {0:s}'.format(key))
+
+  def VerifyStructure(self, parser_mediator, lines):
+    """Verify that this file is a Google Drive Sync log file.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+      lines (str): one or more lines from the text file.
+
+    Returns:
+      bool: True if this is the correct parser, False otherwise.
+    """
+    try:
+      structure = self._GDS_LINE.parseString(lines)
+    except pyparsing.ParseException as exception:
+      logging.debug('Not a Google Drive Sync log file: {0!s}'.format(exception))
+      return False
+
+    try:
+      dfdatetime_time_elements.TimeElementsInMilliseconds(
```

Comment:
Please check the full date and time including the time zone offset.

Reply:
I'm not sure I'm doing this in a sane way -- I just duplicated the datetime-related logic in _ParseRecordLogline for now.

Am I understanding your intent correctly here, and should I make a `_BuildDfdatetimeObject(structure)` method or something to hide the duplication, or is this OK for now and we can tear both pieces out for something cleaner when #47 or related work is done?